src/Modules/OrchardCore.Commerce.Payment.Stripe/Drivers/StripeApiSettingsDisplayDriver.cs

```diff
@@ -77,41 +77,43 @@ public override async Task<IDisplayResult> UpdateAsync(StripeApiSettings section
             return null;
         }
 
-        if (context.GroupId == GroupId)
+        if (context.GroupId != GroupId)
         {
-            var model = new StripeApiSettingsViewModel();
-            var previousSecretKey = section.SecretKey;
-            var previousWebhookKey = section.WebhookSigningSecret;
+            return await EditAsync(section, context);
+        }
+
+        var model = new StripeApiSettingsViewModel();
+        var previousSecretKey = section.SecretKey;
+        var previousWebhookKey = section.WebhookSigningSecret;
+
+        if (await context.Updater.TryUpdateModelAsync(model, Prefix))
```

Comment:
in OC, the new recommendation here put this out side of `if` block. always update the model even if validation failed.